fractions/javaee/undertow/src/main/java/org/wildfly/swarm/undertow/descriptors/WebXmlAsset.java

```diff
@@ -169,14 +182,15 @@ public InputStream openStream() {
         }
 
         for (String eachRole : allRoles) {
-            this.descriptor.getOrCreateSecurityRole()
+            this.descriptor.createSecurityRole()
```

Comment:
Will it create a duplicate role if the role already exists ? (Not sure though such a case is possible :-) )

Reply:
It can't create a duplicate because all security roles are removed from the descriptor few lines above. Also, if there are multiple roles, `getOrCreate` would mean that there's always only one role in the descriptor, so this is actually yet another fix :-)